### PR TITLE
COMP: Update CTK removing code specific to Qt 4

### PR DESCRIPTION
**Pull request by jcfr**
List of CTK changes:

```
$ git shortlog 2b16f9dff..ee5508d6e --no-merges
James Butler (2):
      COMP: Update C++ source files removing code specific to Qt 4
      ENH: Remove custom CTK Dialog options

Jean-Christophe Fillion-Robin (1):
      COMP: Revert removal of custom CTK Dialog options
```

### CODE REVIEW

**Comment by jcfr** [comment]
It turns out that attempting to build against CTK after integrating the following pull request is failing:
* https://github.com/commontk/CTK/pull/1127

Associated error:

```
/home/jcfr/Projects/Slicer-Release/Slicer-build/Modules/Scripted/DICOMLib/Widgets/ui_qSlicerDICOMExportDialog.h:185:64: error: ‘ShowDirsOnly’ is not a member of ‘ctkPathLineEdit’
  185 |         PathLineEdit_OutputFolder->setOptions(ctkPathLineEdit::ShowDirsOnly);
      |                                                                ^~~~~~~~~~~~
```

**Comment by jcfr** [comment]
The issue is that `ctkDirectoryButton` and `ctkPathLineEdit` both derive from `QWidget` and client code uses reference to `ctkPathLineEdit::ShowDirsOnly`

To ensure existing code can still compile, we should revert the removal and instead associated a deprecated warning with the use of the obsolete API.

cc: @jamesobutler

**Comment by jcfr** [comment]
Related pull request:
* https://github.com/commontk/CTK/pull/1128

**Comment by jamesobutler** [comment]
Yes to update the Slicer you’ll need my changes that I have at https://github.com/jamesobutler/Slicer/tree/jamesobutler-ctk-drop-qt4 which I was using when doing my own testing.

I figured it was such a minor change, but I can work to add it as a deprecation on the CTK side.

File: Modules/Scripted/DICOMLib/Widgets/Resources/UI/qSlicerDICOMExportDialog.ui [hunk 1]
```diff
@@ -255,7 +255,7 @@ If unchecked, the exported dataset will be added to the DICOM database.</string>
          <set>ctkPathLineEdit::Dirs|ctkPathLineEdit::Drives|ctkPathLineEdit::Executable|ctkPathLineEdit::NoDot|ctkPathLineEdit::NoDotDot|ctkPathLineEdit::PermissionMask|ctkPathLineEdit::Readable|ctkPathLineEdit::Writable</set>
         </property>
         <property name="options">
-         <set>ctkPathLineEdit::ShowDirsOnly</set>
+         <set>QFileDialog::ShowDirsOnly</set>
```
Comment (jamesobutler) [inline Modules/Scripted/DICOMLib/Widgets/Resources/UI/qSlicerDICOMExportDialog.ui:258]: @jcfr If you don't want this change, you can revert it

Comment (jcfr) [reply Modules/Scripted/DICOMLib/Widgets/Resources/UI/qSlicerDICOMExportDialog.ui]: The proposed change makes sense, I suggest we apply it once we have addressed:
* https://github.com/commontk/CTK/issues/1129

Comment (jcfr) [reply Modules/Scripted/DICOMLib/Widgets/Resources/UI/qSlicerDICOMExportDialog.ui]: > I figured it was such a minor compatibility change, but I can work to add it as a deprecation on the CTK side.

I will do a search to assess if the this is used on the C++ side of Slicer extensions.

The issues is that the `ctkDirectoryButton::Option` and `ctkDirectoryButton::Option` enum values are not available anymore from the class as they derive from `QWidget` and not from `QFileDialog`.

**Pull request merged**